plugins/calibrecompanion.koplugin/main.lua

```diff
@@ -11,6 +12,25 @@ local T = require("ffi/util").template
 
 require("ffi/zeromq_h")
 
+-- supported formats
+local extensions = require("extensions")
+local function getExtensionPathLengths()
+    local t = {}
+    for _, v in pairs(extensions) do
+        t[v] = 37
```

Comment:
What's this 37 ? Some opcode like the another anonymous numbers in this file ? Some real path length, but why 37 (while it was 42 before - but why 42 ? because it's the answer ? so arbitrary ? so many questions :)
Might be worth some comments, as you must have investigated all that, so don't let the knowledge be lost :)

Reply:
It is some magical thing. I really don't know. It could be the maximum file name until the extension or any other stuff. I did change it to 37 because it is what Calibre Companion and Uncaged reported to calibre server. I do trust what CC/uncaged are doing. Plus it was the only difference on the `calibre-debug -g` logs.

It makes sense to notify the server which is the maximum length of a filename, so it can feed books with adjusted lpath.

Reply:
Agreed, probably best to add a `-- magic number from calibre` comment or something. ^_^

Reply:
Sure. Maybe @shermp can shed some light on this issue.

Reply:
I'm afraid this is a magic number to me as well. I just copied whatever CC was using.

Looking at the Calibre source, it looks like this is part of determining max path length. Also does some sums with `PATH_FUDGE_FACTOR`, which in the source, is:
```
    # guess of length of MTP name. The length of the full path to the folder
    # on the device is added to this. That path includes the device's mount point
    # making this number effectively around 10 to 15 larger.
    PATH_FUDGE_FACTOR           = 40
```
Seems to be primarily used here:
```
maxlen = (self.MAX_PATH_LEN - (self.PATH_FUDGE_FACTOR +
                   self.exts_path_lengths.get(dotless_ext, self.PATH_FUDGE_FACTOR)))
```